packages/animation-utils/src/transformation-helpers/interpolate-styles/index.tsx

```diff
@@ -0,0 +1,288 @@
+	) {
+		throw new TypeError(
+			`The start and end values must be of the same type. Start value: ${initialStyleProperty}, end value: ${finalStyleProperty}`,
+		);
+	}
+
+	const initialStylePropertyParts =
+		breakDownValueIntoUnitNumberAndFunctions(initialStyleProperty);
+	const finalStylePropertyParts =
+		breakDownValueIntoUnitNumberAndFunctions(finalStyleProperty);
+
+	if (initialStylePropertyParts.length !== finalStylePropertyParts.length) {
+		throw new TypeError(
+			`The start and end values must have the same structure. Start value: ${initialStyleProperty}, end value: ${finalStyleProperty}`,
+		);
+	}
+
+	const interpolatedValue = initialStylePropertyParts.reduce(
+		(acc, initialStylePropertyPart, index) => {
+			return `${acc} ${interpolatedPropertyPart({
+				inputValue,
+				inputRange,
+				initialStylePropertyPart,
+				finalStylePropertyPart: finalStylePropertyParts[index],
+				initialStyleProperty,
+				finalStyleProperty,
+			})}`;
+		},
+		'',
+	);
+	return interpolatedValue.slice(1);
+};
+
+const interpolateStylesFunction = (
+	inputValue: number,
+	[startingValue, endingValue]: number[],
+	initialStyle: Style,
+	finalStyle: Style,
+): Style => {
+	return Object.keys(initialStyle).reduce((acc, key) => {
+		if (!finalStyle[key as CSSPropertiesKey]) {
+			return {
+				...acc,
+				[key]: initialStyle[key as CSSPropertiesKey],
+			};
+		}
+
+		const finalStyleValue = interpolateProperty(
+			inputValue,
+			[startingValue, endingValue],
+			initialStyle[key as CSSPropertiesKey],
+			finalStyle[key as CSSPropertiesKey],
+		);
+
+		// Avoid number to be a string
+		if (!isNaN(Number(finalStyleValue))) {
+			return {
+				...acc,
+				[key]: Number(finalStyleValue),
+			};
+		}
+
+		return {
+			...acc,
+			[key]: finalStyleValue,
+		};
+	}, {});
+};
+
+function checkInputRange(arr: readonly number[]) {
+	if (arr.length < 2) {
+		throw new Error('inputRange must have at least 2 elements');
+	}
+
+	for (let index = 0; index < arr.length; index++) {
+		if (typeof arr[index] !== 'number') {
+			throw new Error(`inputRange must contain only numbers`);
+		}
+
+		if (arr[index] === -Infinity || arr[index] === Infinity) {
+			throw new Error(
+				`inputRange must contain only finite numbers, but got [${arr.join(
+					',',
+				)}]`,
+			);
+		}
+
+		if (index > 0 && !(arr[index] > arr[index - 1])) {
+			throw new Error(
+				`inputRange must be strictly monotonically non-decreasing but got [${arr.join(
+					',',
+				)}]`,
+			);
+		}
+	}
+}
+
+function checkStylesRange(arr: readonly Style[]) {
+	if (arr.length < 2) {
+		throw new Error('outputStyles must have at least 2 elements');
+	}
+
+	for (const index in arr) {
+		if (typeof arr[index] !== 'object') {
+			throw new Error('outputStyles must contain only objects');
+		}
+	}
+}
+
+/**
+ * @description This function allows you to map a range of values to colors using a concise syntax.
+ * @see [Documentation](https://www.remotion.dev/docs/interpolate-colors)
```

Comment:
This is inaccurate

Reply:
By mistake copied exactly was taking insperation from interpolate color function 😅